### PR TITLE
fix: use items validation for flattened arrays

### PR DESCRIPTION
**Pull request by P0lip**
Addresses https://github.com/stoplightio/platform-internal/issues/5799

image
I'm so happy this code is entirely gone in beta, and that JST has it done more reasonably :see_no_evil: 

`tickets.schema.json` has been updated due to that fix, see the portion of that whole schema document
```json
    "ticketRecipients": {
                            "type": "array",
                            "items": {
                              "type": "string",
                              "enum": [
                                "BOOKER",
                                "CUSTOMER",
                                "PASSENGER",
                                "THIRD_PARTY"
                              ]
                            }
                          },
```

### CODE REVIEW

File: src/tree/utils/walk.ts [hunk 1]
```diff
@@ -49,10 +49,24 @@ export function* processNode(node: JSONSchema4): IterableIterator<SchemaNode> {
       yield combinerNode;
     }
   } else if (type) {
+    const primaryType = getPrimaryType(node);
+    let validationNOde: JSONSchema4 = node;
```
Comment (paulatulis) [inline src/tree/utils/walk.ts:53]: is there a reason this is `validationNOde` and not `validationNode`?

**Review by paulatulis — COMMENTED** [review]
Looks good! I pulled down the branch and tested with the example you show in the summary (thanks for including), and it does what it says it should. I just have one question about the variable name.

File: src/tree/utils/walk.ts (hunk 1 above)
Comment (P0lip) [reply src/tree/utils/walk.ts:53]: Ah yeah. Typo.

**Review by paulatulis — APPROVED** [review]
Bueno

**Pull request merged**